### PR TITLE
capture context for exceptions thrown during MPU commit request

### PR DESCRIPTION
**Pull request by cburroughs**
ref #278

### CODE REVIEW

**Comment by cburroughs** [comment]
This ends up double wrapping during  validation failures, breaking an integration tests. :-(  Working on a revised fix.

**Comment by cburroughs** [comment]
Let me know if there are any concerns about the revised diff.

**Comment by tjcelaya** [comment]
Is there a reason for those two exception types specifically? Not a blocker just wondering why you didn't use more general types like `catch (MantaException | MantaIOException me)`.

**Comment by dekobon** [comment]
They are separate because their inheritance is separate. A `MantaIOException` can be caught by any `catch (IOException e)` clause. Whereas a more generalized MantaException doesn't have to do with IO and a consumer may want to handle all exceptions relating to Manta in a single try/catch clause. What is difficult is that Java doesn't support multiple inheritance and if I remember right you can't catch an interface.

**Comment by tjcelaya** [comment]
**TL;DR** style question was too terse, sounded like a language feature question

I'm aware of the inheritance difference, that's why I combine them in the proposed single-catch. I would assume `catch (MantaException | MantaIOException me) { throw me; }` would achieve the intended effect (rethrowing already-annotated exceptions) and was curious why more specific exception types were selected and had separate catch clauses.

Stepping back to think about it a new kind of exception could be thrown from that `try` and my proposed change would assume that exception is annotated correctly, which might not be the case. This change is more explicit (though the `MantaMultipartException` and `MantaClientHttpResponseException` clauses could still be combined since they do the same thing, but that's more about style).

**Comment by dekobon** [comment]
Fair point.

**Comment by cburroughs** [comment]
Switched to `|`, I'm still forget about the 'new' multiple catch blocks.
